src/state.py

```diff
@@ -11,6 +12,55 @@ def __init__(self):
     def add_student(self, student):
         self.students[student.student_id] = student
 
+    def get_score(self): 
+        score = 0
+        
+        for student in self.students.values(): 
+
+            #checking buddies 
+            for numbers in set(list(student.buddies.values())):
+                    for a_class in set(list(student.buddies.keys())): 
+                        if student.subjects_and_classes[a_class] == self.students[numbers].subjects_and_classes[a_class]: 
+                            score += 30         #the person is at the same class of his friend
+                            student.alone = False
```

Comment:
The flags alone, target and gave_in are never reset. 
This must also be fixed.

Reply:
The flags as they were implemented before (as local variables of the main for loop) were correct. If they are local variables and not properties of each student instance, the problem will be solved. Will make that change

Reply:
> > Please comment the issue that the PR is closing!
> 
> So.... It's just to put at the first comment what is the issue?



> > Please comment the issue that the PR is closing!
> 
> So.... It's just to put at the first comment what is the issue?

@Jumaruba 
Your first "comment" is the pull request description. 
So all you have to do is edit it so it says something like 

> This pull request closes #4